### PR TITLE
Enabling go tests for drivers that ship in Rancher

### PR DESCRIPTION
**Pull request by phillipsj**
Enabled the existing tests for the drivers that we ship in Rancher.

* https://github.com/rancher/windows/issues/143

### CODE REVIEW

**Review by rosskirkpat — APPROVED** [review]

File: drivers/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -651,7 +648,9 @@ func (d *Driver) Create() error {
 
 	if err := d.innerCreate(); err != nil {
 		// cleanup partially created resources
-		d.Remove()
+		if err := d.Remove(); err != nil {
+			return err
+		}
```
Comment (thedadams) [inline drivers/amazonec2/amazonec2.go:653]: If there was an error during create, then we should also report that. It seems reasonable to me that if there was an error during create, then there would also be an error during remove.

File: scripts/ci [hunk 2]
```diff
@@ -6,4 +6,5 @@ cd $(dirname $0)
 ./clean
 # ./validate
```
Comment (thedadams) [inline scripts/ci:7]: Should we also turn on the validation?

File: drivers/amazonec2/amazonec2_test.go [hunk 3]
```diff
@@ -116,44 +106,6 @@ func TestConfigureSecurityGroupPermissionsSkipReadOnly(t *testing.T) {
 	assert.Len(t, perms, 0)
 }
 
-func TestConfigureSecurityGroupPermissionsOpenPorts(t *testing.T) {
```
Comment (thedadams) [inline drivers/amazonec2/amazonec2_test.go:119]: Should we test that the Rancher security group has the expected open ports?

**Review by thedadams — COMMENTED** [review]
I was in the middle of my review when this PR was merged. I was going to request the following changes.

**Pull request merged**